src/Compilation.zig

```diff
@@ -4138,6 +4157,12 @@ pub fn stage1AddLinkLib(comp: *Compilation, lib_name: []const u8) !void {
     if (comp.bin_file.options.skip_linker_dependencies) return;
 
     // This happens when an `extern "foo"` function is referenced by the stage1 backend.
+    if (comp.getTarget().os.tag == .wasi and mem.eql(u8, "wasi_snapshot_preview1", lib_name)) {
+        // Any referenced symbol from this lib, will be undefined until
+        // runtime as this lib is provided directly by the runtime.
+        return;
+    }
```

Comment:
I don't quite understand why this is here. The comment doesn't make sense to me because that is true about all dynamically linked libraries. What makes `wasi_snapshot_preview1` special?

Reply:
`wasi_snapshot_preview1` is actually not a dynamic library per se; it's a set of syscalls implicitly provided by the sandbox, therefore we cannot really link against it. There are two options here: we either remove any mention of `wasi_snapshot_preview1` from the linker line (which will be there since we import syscalls directly into our libstd), or disable linking dynamically by `wasm-ld` (which was the case until now). I'm happy either way, and if you think we should preserve the status quo, I'll delete this bit, and remove any mention of dynamic linking from `wasm-ld` wrapper in `Wasm.zig`.

Reply:
I think this logic would make more sense to me if it was in link/Wasm.zig which sounds like one of your suggestions :+1:

Reply:
Oh, actually. I'll move this logic into `link/Wasm.zig` but I cannot get rid of the logic which unpacks `system_libs` since we are allowed to pass static libs via `-l` flags.